workspaces/libnpmexec/lib/index.js

```diff
@@ -86,20 +85,47 @@ const exec = async (opts) => {
     packages.push(args[0])
   }
 
+  // figure out whether we need to install stuff, or if local is fine
+  const localArb = new Arborist({
+    ...flatOptions,
+    path,
+  })
+  const localTree = await localArb.loadActual()
+
+  const getLocalManifest = ({ tree, name }) => {
+    // look up the package name in the current tree inventory,
+    // if it's found then return that normalized pkg data
+    let node
+    const nodes = tree.inventory.query('packageName', name)
+    for (const n of nodes) {
+      node = n
+      break
+    }
+
+    if (node) {
+      return {
+        _id: node.pkgid,
+        ...node.package,
+        [_localManifest]: true,
+      }
+    }
+  }
+
   // If we do `npm exec foo`, and have a `foo` locally, then we'll
   // always use that, so we don't really need to fetch the manifest.
   // So: run npa on each packages entry, and if it is a name with a
-  // rawSpec==='', then try to readPackageJson at
-  // node_modules/${name}/package.json, and only pacote fetch if
-  // that fails.
+  // rawSpec==='', then try to find that node name in the tree inventory
+  // and only pacote fetch if that fails.
   const manis = await Promise.all(packages.map(async p => {
     const spec = npa(p, path)
     if (spec.type === 'tag' && spec.rawSpec === '') {
-      // fall through to the pacote.manifest() approach
-      try {
-        const pj = resolve(path, 'node_modules', spec.name, 'package.json')
-        return await readPackageJson(pj)
-      } catch (er) {}
+      const localManifest = getLocalManifest({
+        tree: localTree,
+        name: spec.name,
+      })
+      if (localManifest) {
+        return localManifest
```

Comment:
`getLocalManifest()` will return `undefined` if no manifest is found, right? so why not skip the condition and always return `localManifest` here since it's the same effect either way?

Reply:
oops, I believe this is just a leftover from previous of moving the logic to `getLocalManifest` 😬 good catch!